qa-tests-backend/src/test/groovy/ProcessesListeningOnPortsTest.groovy

```diff
@@ -254,6 +256,81 @@ class ProcessesListeningOnPortsTest extends BaseSpecification {
         destroyDeployments()
     }
 
+    def "Verify listening endpoints for collector are reported"() {
+        given:
+
+        String collectorUid = orchestrator.getDaemonSetId(new DaemonSet(name: "collector", namespace: "stackrox"))
+        log.info "collectorUid= ${collectorUid}"
+
+        def processesListeningOnPorts = evaluateWithRetry(10, 10) {
+                def temp = ProcessesListeningOnPortsService
+                        .getProcessesListeningOnPortsResponse(collectorUid)
+                return temp
+        }
+
+        // First check that the listening endpoint appears in the API
+        assert processesListeningOnPorts
+
+        def list = processesListeningOnPorts.listeningEndpointsList
+        // The size of the list depends upon the number of colletors
+        // which can vary based upon the environment
+        assert list.size() > 1
+
+        def endpoint1 = list.find { it.endpoint.port == 8080 }
+
+        assert endpoint1
+        assert endpoint1.deploymentId
+        assert endpoint1.podId
+        assert endpoint1.podUid
+        assert endpoint1.clusterId
+        assert endpoint1.Namespace
+        assert endpoint1.containerName == "collector"
+        assert endpoint1.signal.name == "collector"
+        assert endpoint1.signal.execFilePath == "/usr/local/bin/collector"
+
+        def endpoint2 = list.find { it.endpoint.port == 9090 }
+
+        assert endpoint2
+        assert endpoint2.deploymentId
+        assert endpoint2.podId
+        assert endpoint2.podUid
+        assert endpoint2.clusterId
+        assert endpoint2.Namespace
+        assert endpoint2.containerName == "collector"
+        assert endpoint2.signal.name == "collector"
+        assert endpoint2.signal.execFilePath == "/usr/local/bin/collector"
+
+        def pagination = new Pagination(1, 0)
```

Comment:
how about moving pagination to a param? Also I'm not sure if this should be tested with e2e. Why not testing it in go unit test?
https://blog.jetbrains.com/idea/2021/02/tutorial-spock-part-3-data-driven-testing/#data-tables

Reply:
I have added unit tests. I like the e2e tests. One reason is that it checks the collector listening endpoints, which was not done before. I have removed all usage of `evaluateWithRetry` now that I am using the `@Retry` annotation. I think that makes your comment about moving pagination to a param, if I understood it correctly.

Reply:
I separated out the test for checking collector listening endpoints without pagination from the now parameterized test involving pagination.